### PR TITLE
[v1.1.x-backport] backport integration tests framework fixes

### PR DESCRIPTION
**Pull request by meldafrawi**
Signed-off-by: Mohamed Eldafrawi <mohamed.eldafrawi@suse.com>

### CODE REVIEW

File: test_framework/Dockerfile.setup [hunk 1]
```diff
@@ -2,7 +2,7 @@ From alpine:latest
 
 ARG KUBECTL_VERSION=v1.20.2
 
-ARG RKE_VERSION=v1.2.9
+ARG RKE_VERSION=v1.3.4
```
Comment (khushboo-rancher) [inline test_framework/Dockerfile.setup:5]: RKE v1.3.4 does include `v1.22.5-rancher1-1` K8s version, what's the default one as the v1.1.x will not work on v1.22.5?

File: test_framework/terraform/aws/centos/main.tf [hunk 2]
```diff
@@ -367,7 +374,7 @@ resource "null_resource" "wait_for_docker_start_worker" {
   }
 }
 
-# Download KUBECONFIG file (for k3s arm64 only)
+# Download KUBECONFIG file (for k3s k3s only)
```
Comment (khushboo-rancher) [inline test_framework/terraform/aws/centos/main.tf:377]: Is this a typo?

File: test_framework/terraform/aws/centos/variables.tf [hunk 3]
```diff
@@ -88,14 +88,24 @@ variable "lh_aws_instance_root_block_device_size_worker" {
...
+  default     = "k3s"
+  description = "kubernetes distro version to install [rke, k3s]  (default: k3s)"
 }
 
-variable "k3s_version" {
+variable "k8s_distro_version" {
   type        = string
-  default     = "v1.20.8+k3s1"
-  description = "K3s version that will be deployed in case of arch=arm64"
+  default     = "v1.23.1+k3s2"
```
Comment (khushboo-rancher) [inline test_framework/terraform/aws/centos/variables.tf:99]: Longhorn on branch v1.1.x doesn't support v1.23

File: test_framework/terraform/aws/oracle/variables.tf [hunk 4]
```diff
@@ -89,14 +88,18 @@ variable "lh_aws_instance_root_block_device_size_worker" {
...
+  default     = "k3s"
+  description = "kubernetes distro version to install [rke, k3s]  (default: k3s)"
 }
 
-variable "k3s_version" {
+variable "k8s_distro_version" {
   type        = string
-  default     = "v1.20.8+k3s1"
-  description = "K3s version that will be deployed in case of arch=arm64"
+  default     = "v1.23.1+k3s2"
```
Comment (khushboo-rancher) [inline test_framework/terraform/aws/oracle/variables.tf:99]: Version correction as above.

File: test_framework/terraform/aws/rhel/user-data-scripts/provision_rke.sh.tpl [hunk 5]
```diff
@@ -0,0 +1,24 @@
...
+sudo yum install -y iscsi-initiator-utils nfs-utils nfs4-acl-tools
+sudo systemctl -q enable iscsid
+sudo systemctl start iscsid
+
+until (curl https://releases.rancher.com/install-docker/$${DOCKER_VERSION}.sh | sudo sh); do
+  echo 'docker did not install correctly'
+  sleep 2
+done
+
+sudo usermod -aG docker centos
```
Comment (khushboo-rancher) [inline test_framework/terraform/aws/rhel/user-data-scripts/provision_rke.sh.tpl:24]: Is the user `centos` or `ec2-user`?

**Review by khushboo-rancher — COMMENTED** [review]
LGTM in general, 
Just a doubt about the `kubernetes version` and some minor comments.

File: test_framework/Dockerfile.setup (hunk 1 above)
Comment (meldafrawi) [reply test_framework/Dockerfile.setup:5]: RKE v1.3.4 include the following k8s versions:
image

However, I've fixed the default to be `v1.21.8-rancher1-1`

File: test_framework/terraform/aws/centos/main.tf (hunk 2 above)
Comment (meldafrawi) [reply test_framework/terraform/aws/centos/main.tf]: Yes, Fixed, Thanks.

File: test_framework/terraform/aws/centos/variables.tf (hunk 3 above)
Comment (meldafrawi) [reply test_framework/terraform/aws/centos/variables.tf]: Fixed, Thanks.

File: test_framework/terraform/aws/oracle/variables.tf (hunk 4 above)
Comment (meldafrawi) [reply test_framework/terraform/aws/oracle/variables.tf]: Fixed, Thanks.

File: test_framework/terraform/aws/rhel/user-data-scripts/provision_rke.sh.tpl (hunk 5 above)
Comment (meldafrawi) [reply test_framework/terraform/aws/rhel/user-data-scripts/provision_rke.sh.tpl]: Fixed, Thanks

**Review by khushboo-rancher — APPROVED** [review]

**Pull request merged**